host/sgx/enter.c

```diff
@@ -156,14 +159,21 @@ void oe_enter(
     uint64_t* arg4,
     oe_enclave_t* enclave)
 {
-    // The general purpose registers are preserved by the compiler.
-    // The floating point state and the flags must be explicitly preserved.
-    // The space for saving the floating-point state must be 16 byte aligned.
-    OE_ALIGNED(16)
-    uint64_t fx_state[64];
+    // Additional control registers that need to be preserved as part of the
+    // Windows and Linux x64 ABIs
+    uint32_t mxcsr = 0;
+    uint16_t fcw = 0;
+
     oe_ecall_context_t ecall_context = {{0}};
     _setup_ecall_context(&ecall_context);
 
+    // Compiler will usually handle this on exiting a function that uses AVX,
+    // but we need to avoid the AVX-SSE transition penalty here manually as
+    // part of the transition to enclave. See https://software.intel.com/content
+    // /www/us/en/develop/articles/avoiding-avx-sse-transition-penalties.html
+    if (oe_is_avx_enabled)
+        asm volatile("vzeroupper \n\t" : : :);
```

Comment:
Since vzeroupper wipes out the top 128 bits of the ymm registers, we'd need to specify the ymm registers in the clobber list to be correct.

Reply:
Similar to the flags, this doesn't seem to change the generated assembly, but I've added the YMM registers to the clobber list.